### ISSUE TITLE
`awscdk` target requires a global install of `aws-cdk-lib`

### ISSUE THREAD

**Issue by Chriscbr** [winglang/wing#2478]
### I tried this:

My code:
```js
bring cloud;

let counter = new cloud.Counter();

new cloud.Function(inflight () => {
  let val = counter.dec(2);
  log("${val}");
});
```

### This happened:

I tried compiling, saw the error, installed the library (and I see it in my package.json) and then retried compiling it:
```
$ wing compile -t awscdk counter.w
Cannot find module 'aws-cdk-lib'. The target "awscdk" requires this module to be installed.
$ npm i aws-cdk-lib

changed 3 packages in 9s

10 packages are looking for funding
  run `npm fund` for details
$ wing compile -t awscdk counter.w
Cannot find module 'aws-cdk-lib'. The target "awscdk" requires this module to be installed.
```

### I expected this:

No error

### Is there a workaround?

Install the package globally (with `npm i -g`) instead of locally.

### Component

CLI

### Wing Version

0.16.3

### Node.js Version

18.14.1

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by eladb** [comment]
Here's the stack trace (`DEBUG=1`):

```
Error: Cannot find module 'aws-cdk-lib'
Require stack:
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/target-awscdk/app.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/app.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/core/index.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/cloud/api.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/cloud/index.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/node_modules/@winglang/sdk/lib/index.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/dist/commands/test.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/dist/commands/index.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/dist/cli.js
- /Users/eladb/.volta/tools/image/packages/winglang/lib/node_modules/winglang/bin/wing
```

I think the root cause is that the preflight module resolution is done from globally installed CLI, so it doesn't find the locally installed `aws-cdk-lib`.

If I install `winglang` locally, it works:

```sh
$ npm i winglang aws-cdk-lib
$ wing compile -t awscdk hello.w
<WORKS!>
```

Alternatively, installing `aws-cdk-lib` globally also works:

```sh
$ npm i -g aws-cdk-lib
```

So we have a workaround (added to the issue summary) and added a label. The solution is to add `cwd` to the module preflight resolution, which could be tricky with the node VM.

I suggest that for now, we will just change the message to tell the user to use `npm i -g` if they use the globally installed CLI, ok?

**Comment by Chriscbr** [comment]
> So we have a workaround (added to the issue summary) and added a label. The solution is to add `cwd` to the module preflight resolution, which could be tricky with the node VM.
> 
> I suggest that for now, we will just change the message to tell the user to use `npm i -g` if they use the globally installed CLI, ok?

Thanks for finding the workaround 🙏 .  Yes I think that solution is okay for now...

**Comment by staycoolcall911** [comment]
Thanks @eladb. 
In light of this - changing the issue from p1 to p2.

**Comment by ekeren** [comment]
For now, I've changed the error message to include the workaround (globally)

https://github.com/winglang/wing/pull/2624

**Comment by eladb** [comment]
This is current error we are seeing:

```
Error: ERROR: Cannot find module 'aws-cdk-lib'. The target "awscdk" requires this module to be installed globally (using "npm i -g").
```

**Comment by eladb** [comment]
Changed the title and moving to P2

**Comment by staycoolcall911** [comment]
@marciocadev , @Chriscbr - can we close this issue?

**Comment by marciocadev** [comment]
No @staycoolcall911 

Still need global install

**Comment by staycoolcall911** [comment]
Should be resolved after #3620 is done

**Comment by staycoolcall911** [comment]
Global install is not required anymore for awscdk
